libmscore/layout.cpp

```diff
@@ -841,7 +841,7 @@ void Score::layoutChords3(std::vector<Note*>& notes, const Staff* staff, Segment
       for (int i = nNotes-1; i >= 0; --i) {
             Note* note     = notes[i];
             Accidental* ac = note->accidental();
-            if (ac && !note->fixed()) {
+            if (ac && ac->addToSkyline() && !note->fixed()) {
```

Comment:
This doesn't work correctly - it causes the layout to be skipped entirely.  That means, among other things, the invisible or non-autoplaced accidental will disappear completely on save/reload.

Dealing with this will be tricky because of the interdependencies between accidentals on a single chord (which is what this code - some of the very first I contributed to MuseScore! - is dealing with).  Ideally we want to still place the accidental where it would have been otherwise if it were visible and autoplaced, but we would not let this affect the position of other accidentals.  But I'm not sure how feasible that is here.

The easiest answer would be to lose this particular change.  This would be fine for chords with a single accidental. 
 But the downside would be, a single invisible accidental in the middle of a chord with multiple accidentals would leave a "gap" in the accidental stack, just as it does now.

Another possibility is to *start* the block so we still get the layout, but add an "if (!ac->visible()) continue;" right after the layout.  Well, might be good to set some explicit x position, otherwise it will end up staying where it is when you first make it invisible, but then become 0 on save/reload.  This wouldn't be acceptable for visible accidentals with autoplace turned off, though - they still need to stack normally.

Anything else would probably require deeper change to the accidental stacking algorithm further below, and I'm not even sure it would be possible to define a more sensible behavior.